pkg/performanceprofile/controller/performanceprofile/hypershift/components/handler.go

```diff
@@ -0,0 +1,314 @@
+package components
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	mcov1 "github.com/openshift/api/machineconfiguration/v1"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/config"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/manifestset"
+	profileutil "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components/profile"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/hypershift"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/resources"
+)
+
+const (
+	hypershiftPerformanceProfileNameLabel = "hypershift.openshift.io/performanceProfileName"
+	hypershiftNodePoolLabel               = "hypershift.openshift.io/nodePool"
+	tunedConfigMapLabel                   = "hypershift.openshift.io/tuned-config"
```

Comment:
nit: ok, REALLY a nit but still: we use `snakeCase` for the previous two labels and `dash-separated` for this one. Is it possible to be consistent? do we have backward compatibility constraints?

Reply:
These were taken verbatim from the old PR that Jose did and are also appears in the original enhancement:
https://github.com/openshift/enhancements/blob/4668a0825c59739dfafd2ae661c16cf30f540946/enhancements/hypershift/performanceprofilecontroller-node-tuning.md?plain=1#L109-#L111

Although the argument of mixing styles is completely in place and I agree with it, the cost of changing the EP + the code on hypershift side is too high